pkgs/by-name/mu/multiqc/package.nix

```diff
@@ -0,0 +1,115 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  procps,
+  versionCheckHook,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "multiqc";
+  version = "1.26";
+
+  disabled = python3Packages.pythonOlder "3.8";
+
+  # Two data sources. One for the code, another for the test data
+  srcs = [
+    (fetchFromGitHub {
+      name = "multiqc";
+      owner = "MultiQC";
+      repo = "MultiQC";
+      tag = "v${version}";
+      hash = "sha256-MPAw6gG/3LzdskkDXOTDEM1NpG0sH9GvklYFQ1ZXWIs=";
+    })
+    (fetchFromGitHub {
+      owner = "MultiQC";
+      repo = "test-data";
+      rev = "67435083a8bfa228dca3dda7d835facef15fc2c7";
+      hash = "sha256-oYmPIJSy6dOKPcMr3B4foGoWcerA29x0XeGoU4dSYsA=";
+      name = "test-data";
+    })
+  ];
+
+  sourceRoot = "multiqc";
+
+  # Some tests run subprocess.run() with "ps"
+  nativeBuildInputs = [
+    procps
+  ];
+
+  # Some tests run subprocess.run() with "multiqc"
+  preCheck = ''
+    chmod -R u+w ../test-data
+    ln -s ../test-data .
+    export PATH=$PATH:$out/bin
+  '';
```

Comment:
This step is related to the tests/checks, therefore it should be after the declaration of the dependencies (`dependencies`, `optional-dependencies`).

The idea here is to group attributes logically so that when you read the derivation, it's easy to understand and in the right order.